### PR TITLE
2 licence status calculator

### PR DESCRIPTION
**Pull request by atodd-noms**

### CODE REVIEW

File: server/utils/licenceStatus.js [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        case licenceStages.ELIGIBILITY:
+            return [getEligibilityStageState(data)];
+        default:
+            return [];
+    }
+}
+
+const combiner = (acc, data) => {
+    Object.assign(acc.tasks, {...data.tasks});
+    Object.assign(acc.decisions, {...data.decisions});
```
Comment (whitfield-mj) [inline server/utils/licenceStatus.js]: {...data.tasks} is equivalent to data.tasks

Comment (whitfield-mj) [reply server/utils/licenceStatus.js]: Do you think this is more readable? Seems more in-keeping with what we do elsewhere
`const combiner = (acc, data) => {
    const combinedTasks = {...acc.tasks, ...data.tasks};
    const combinedDecisions = {...acc.decisions, ...data.decisions};

    return {
        ...acc,
        tasks: combinedTasks,
        decisions: combinedDecisions
    };
};
`

File: server/utils/licenceStatus.js [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+            return [getEligibilityStageState(data), getRoStageState(data), getApprovalStageState(data)];
+        case licenceStages.PROCESSING_CA:
+        // fall through
+        case licenceStages.PROCESSING_RO:
+            return [getEligibilityStageState(data), getRoStageState(data)];
+        case licenceStages.ELIGIBILITY:
+            return [getEligibilityStageState(data)];
+        default:
+            return [];
+    }
```
Comment (whitfield-mj) [inline server/utils/licenceStatus.js]: Don't object to this method. If it were me, I'd do something like: 
```
function getRequiredState(stage, data) {

    const config = {
        [licenceStages.DECIDED]: [getEligibilityStageState, getRoStageState, getApprovalStageState],
        [licenceStages.APPROVAL]: [getEligibilityStageState, getRoStageState, getApprovalStageState],
        [licenceStages.PROCESSING_CA]: [getEligibilityStageState, getRoStageState],
        [licenceStages.PROCESSING_RO]: [getEligibilityStageState, getRoStageState],
        [licenceStages.ELIGIBILITY]: [getEligibilityStageState],
    };

    return config[stage].map(getStateMethod => getStateMethod(data));
}

```

Comment (atodd-noms) [reply server/utils/licenceStatus.js]: Yeah that's nicer

File: server/utils/licenceStatus.js (hunk 1 above)
Comment (atodd-noms) [reply server/utils/licenceStatus.js]: I had something like that a one point. Have applied this change

**Pull request merged**